Improvement: Signature tests don't work on newer Java versions due to `TCKArchiveProcessor` looking for specific versions

### Specification

Signature test failures

### Assertion

java.lang.RuntimeException: No signature file exists for name: jakarta.enterprise.concurrent.sig_25

### TCK Version

3.1.1

### Implementation being tested

WildFly/Red Hat

### Challenge Scenario

Something else

### Full Description

I'm not certain I'm doing this correctly. The [`TCKArchiveProcessor`](https://github.com/jakartaee/concurrency/blob/1f3f747ea30700a1677376e68ca354b705ea2eea/tck/src/main/java/ee/jakarta/tck/concurrent/framework/arquillian/extensions/TCKArchiveProcessor.java#L91) looks for a signature test file based on the `java.specification.version`. If you run on any version of Java other than 17 or 21 the signature test will fail with:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.007 s <<< FAILURE! -- in ee.jakarta.tck.concurrent.spec.signature.SignatureTests
[ERROR] ee.jakarta.tck.concurrent.spec.signature.SignatureTests -- Time elapsed: 0.007 s <<< ERROR!
java.lang.RuntimeException: No signature file exists for name: jakarta.enterprise.concurrent.sig_25
	at ee.jakarta.tck.concurrent.framework.arquillian.extensions.TCKArchiveProcessor.appendSignaturePackages(TCKArchiveProcessor.java:94)
	at ee.jakarta.tck.concurrent.framework.arquillian.extensions.TCKArchiveProcessor.process(TCKArchiveProcessor.java:56)
	at org.jboss.arquillian.container.test.impl.client.deployment.DeploymentGenerator.applyApplicationProcessors(DeploymentGenerator.java:274)
	at org.jboss.arquillian.container.test.impl.client.deployment.DeploymentGenerator.buildTestableDeployments(DeploymentGenerator.java:217)
	at org.jboss.arquillian.container.test.impl.client.deployment.DeploymentGenerator.createTestableDeployments(DeploymentGenerator.java:149)
	at org.jboss.arquillian.container.test.impl.client.deployment.DeploymentGenerator.generateDeployment(DeploymentGenerator.java:90)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:102)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:89)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:61)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.execute(ContainerEventController.java:94)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:102)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:89)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:82)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:94)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:68)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:94)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.beforeClass(EventTestRunnerAdaptor.java:88)
	at org.jboss.arquillian.junit5.ArquillianExtension.beforeAll(ArquillianExtension.java:44)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
```

There is a hack to work around this by setting the `java.specification.version` to a known version, however that feels wrong.

With the newer sigtest-maven-plugin we should be able to only provide a single file that works on all versions of the currently released Java Runtime Environments.

### Additional Context

_No response_

### Is there an existing challenge for this?

- [x] I have searched the existing issues

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/framework/arquillian/extensions/TCKArchiveProcessor.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/framework/arquillian/extensions/TCKArchiveProcessor.java
@@ -35,7 +35,6 @@ import org.jboss.shrinkwrap.impl.base.asset.AssetUtil;
 import org.jboss.shrinkwrap.resolver.api.maven.Maven;
 
 import ee.jakarta.tck.concurrent.common.signature.ConcurrencySignatureTestRunner;
-import ee.jakarta.tck.concurrent.framework.TestProperty;
 import ee.jakarta.tck.concurrent.framework.junit.anno.Common;
 import ee.jakarta.tck.concurrent.framework.junit.anno.Signature;
 
@@ -85,7 +84,15 @@ public class TCKArchiveProcessor implements ApplicationArchiveProcessor {
         if (!testClass.isAnnotationPresent(Signature.class)) {
             return; //Nothing to append
         }
-        final String jdkVersion = TestProperty.javaSpecVer.getValue();
+        final String jdkVersion;
+        final int currentJdkVersion = Runtime.version().feature();
+        if (currentJdkVersion >= 25) {
+            jdkVersion = "25";
+        } else if (currentJdkVersion >= 21) {
+            jdkVersion = "21";
+        } else {
+            throw new RuntimeException("Unsupported Java version %d".formatted(currentJdkVersion));
+        }
         
         final Package signaturePackage = ConcurrencySignatureTestRunner.class.getPackage();
         final String signatureFileName = ConcurrencySignatureTestRunner.SIG_FILE_NAME + "_" + jdkVersion;

